### ISSUE TITLE
Excessive computation time for matrix construction with gradients in bilinear form in Julia 1.10.1

### ISSUE THREAD

**Issue by Galaxxy31** [bcube-project/Bcube.jl#47]
After switching to Julia 1.10.1, I noticed that the execution time of the `assemble_bilinear` function had greatly increased when the bilinear form contained gradients.
Here's a working example:

```Julia
using Bcube
using LinearAlgebra
using TimerOutputs

const degree = 1
const degquad = 2 * degree + 1

to = TimerOutput()

mesh = rectangle_mesh(251,11 ; xmax = 1.0, ymax = 0.1)
Uspace = TrialFESpace(FunctionSpace(:Lagrange, degree), mesh; size=2)
Vspace = TestFESpace(Uspace)
dΩ = Measure(CellDomain(mesh), degquad)

a(u,v) = ∫( u ⋅ v )dΩ
m(u,v) = ∫( ∇(u) ⋅ ∇(v) )dΩ
A = assemble_bilinear(a, Uspace, Vspace)
M = assemble_bilinear(m, Uspace, Vspace)

for i in 1:10
    @timeit to "build basic" begin
        assemble_bilinear(a, Uspace, Vspace)
    end
    @timeit to "build grad" begin
        assemble_bilinear(m, Uspace, Vspace)
    end
end

show(to)
```
In this example run on my machine, under Julia 1.8.2, the "basic" matrix builds in ~50ms and the "gradient" matrix builds in ~150ms. Under Julia 1.10.1, the former also builds in ~50ms, but the latter takes ~3s.

**Comment by ghislainb** [comment]
For the record, when I run the following lines:
```julia
@btime assemble_bilinear($a, $Uspace, $Vspace)
@btime assemble_bilinear($m, $Uspace, $Vspace)
```
I get:

- Julia 1.9.3 (on my laptop):                  
```
38.904 ms (245117 allocations: 139.03 MiB)     
134.057 ms (1655117 allocations: 365.49 MiB) 
```
- Julia 1.9.1 (on my older workstation) 
```
81.845 ms (245117 allocations: 136.76 MiB)    
322.450 ms (1655117 allocations: 365.49 MiB)
```

**Comment by ghislainb** [comment]
Solved in #62:
```julia
 13.339 ms (117 allocations: 13.82 MiB)
 56.614 ms (117 allocations: 13.82 MiB)
```
_(with Julia v1.10.1)_
